core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -342,7 +342,7 @@ class LogSegment(val log: FileMessageSet,
    * The time this segment has waited to be rolled. If the first message in the segment does not have a timestamp,
    * the time is based on the create time of the segment. Otherwise the time is based on the timestamp of that message.
    */
-  def timeWaitedForRoll(now: Long) : Long= {
+  def timeWaitedForRoll(now: Long, messageTimestamp: Long) : Long= {
```

Comment:
Since we often don't need the `now` at all, would it make sense to do it via the `time` instance in the class instead of a parameter?